app/src/modules/trigger/trigger.c

```diff
@@ -9,101 +9,372 @@
+
+	/* Time in real-time mode */
+	int64_t time_in_real_time_mode_ms;
+
+} state_object;
+
+/* Handler called when the RTM duration timer expires */
+static void rtm_duration_trigger_handler(struct k_timer * timer_id)
 {
+	ARG_UNUSED(timer_id);
+
+	int unused = 0;
 	int err;
-	int not_used = -1;
-	bool fota_ongoing = true;
 
-	err = zbus_chan_read(&FOTA_ONGOING_CHAN, &fota_ongoing, K_NO_WAIT);
+	LOG_DBG("RTM duration timer expired");
+
+	err = zbus_chan_pub(&PRIV_TIMER_CHAN, &unused, K_NO_WAIT);
 	if (err) {
-		LOG_ERR("zbus_chan_read, error: %d", err);
+		LOG_ERR("zbus_chan_pub, error: %d", err);
 		SEND_FATAL_ERROR();
 		return;
 	}
+}
 
-	if (fota_ongoing) {
-		LOG_DBG("FOTA ongoing, skipping trigger message");
-		return;
-	}
+/* Delayed work used to signal triggers to the rest of the system */
+static void trigger_work_fn(struct k_work *work)
+{
+	int err;
+	int not_used = -1;
 
 	LOG_DBG("Sending trigger message");
 
-	err = zbus_chan_pub(&TRIGGER_CHAN, &not_used, K_SECONDS(MSG_SEND_TIMEOUT_SECONDS));
+	err = zbus_chan_pub(&TRIGGER_CHAN, &not_used, K_SECONDS(1));
 	if (err) {
 		LOG_ERR("zbus_chan_pub, error: %d", err);
 		SEND_FATAL_ERROR();
 		return;
 	}
 
-	k_work_reschedule(&trigger_work, update_interval);
+	k_work_reschedule(&trigger_work, K_SECONDS(state_object.update_interval_current_sec));
 }
 
+/* Button handler called when a user pushes a button */
 static void button_handler(uint32_t button_states, uint32_t has_changed)
 {
-	if (has_changed & button_states) {
-		trigger_work_fn(NULL);
+	int err;
+	uint8_t button_number = 1;
+
+	if (has_changed & button_states & DK_BTN1_MSK) {
+		LOG_DBG("Button 1 pressed!");
+
+		err = zbus_chan_pub(&BUTTON_CHAN, &button_number, K_SECONDS(1));
+		if (err) {
+			LOG_ERR("zbus_chan_pub, error: %d", err);
+			SEND_FATAL_ERROR();
+			return;
+		}
+	}
+}
+
+/* Convenience timer used to start/restart the RTM duration timer */
+static void rtm_duration_timer_start(void)
+{
+	LOG_DBG("Starting RTM duration timer: %d seconds", REAL_TIME_DURATION_INTERVAL_SEC);
+
+	k_timer_start(&rtm_duration_timer, K_SECONDS(REAL_TIME_DURATION_INTERVAL_SEC), K_NO_WAIT);
+}
+
+/* Zephyr State Machine framework handlers */
+
+/* HSM states:
+ *
+ * STATE_INIT: Initializing module
+ * STATE_CONNECTED: Connected to cloud. Sending triggers on timers and button presses
+ *	- STATE_REAL_TIME_MODE: Sending triggers every 10 seconds for 10 minutes
+ *	- STATE_NORMAL_MODE: Sending triggers every configured update interval
+ * STATE_DISCONNECTED: Sending of triggers are blocked
+ */
+
+/* STATE_INIT */
+
+static void init_entry(void *o)
+{
+	ARG_UNUSED(o);
+
+	LOG_DBG("init_entry");
+
+	int err = dk_buttons_init(button_handler);
+
+	if (err) {
+		LOG_ERR("dk_buttons_init, error: %d", err);
+		SEND_FATAL_ERROR();
+		return;
+	}
+}
+
+static void init_run(void *o)
+{
+	struct s_object *user_object = o;
+
+	LOG_DBG("init_run");
+
+	if ((user_object->status == CLOUD_CONNECTED) && user_object->chan == &CLOUD_CHAN) {
```

Comment:
```suggestion
	if ((user_object->chan == &CLOUD_CHAN) && (user_object->status == CLOUD_CONNECTED)) {
```
Suggest to verify the channel before checking the status.

I just came to think about the updated behavior that I'm working on in the transport module doesn't quite fit here. 
`CLOUD_CONNECTED` will be sent upon the first connection establishment to cloud, so that's fine.
However, when network is lost and data can't be sent, the cloud will still be connected because of CID in use, ie there will be no `CLOUD_DISCONNECTED`.

So this module will continue to issue triggers. Might have to add a `CLOUD_PAUSED` event or something to send from the transport module.

Reply:
_Suggest to verify the channel before checking the status._

Good idea